x-pack/plugins/security/server/config.ts

```diff
@@ -385,29 +389,16 @@ export function createConfig(
 }
 
 function getSessionConfig(session: RawConfigType['session'], providers: ProvidersConfigType) {
-  const defaultAnonymousSessionLifespan = schema.duration().validate('30d');
   return {
     cleanupInterval: session.cleanupInterval,
     getExpirationTimeouts({ type, name }: AuthenticationProvider) {
       // Both idle timeout and lifespan from the provider specific session config can have three
       // possible types of values: `Duration`, `null` and `undefined`. The `undefined` type means that
       // provider doesn't override session config and we should fall back to the global one instead.
       const providerSessionConfig = providers[type as keyof ProvidersConfigType]?.[name]?.session;
-
-      // We treat anonymous sessions differently since users can create them without realizing it. This may lead to a
-      // non controllable amount of sessions stored in the session index. To reduce the impact we set a 30 days lifespan
-      // for the anonymous sessions in case neither global nor provider specific lifespan is configured explicitly.
-      // We can remove this code once https://github.com/elastic/kibana/issues/68885 is resolved.
-      const providerLifespan =
-        type === 'anonymous' &&
-        providerSessionConfig?.lifespan === undefined &&
-        session.lifespan === undefined
-          ? defaultAnonymousSessionLifespan
-          : providerSessionConfig?.lifespan;
```

Comment:
So now that we are removing this, what happens if a user sets `xpack.security.session.lifespan: 0` and enables anonymous access? Won't we still have this problem?

Reply:
Yeah, but it's like this in `master` too, right? I mean if I set this setting to `0` (or `null`) globally then I'd expect all providers to obey where it makes sense (e.g. not for idle timeout that doesn't make much sense for anonymous provider).

Reply:
> Yeah, but it's like this in `master` too, right?
I assume you mean `7.x`?
Well yeah you could explicitly set session lifespan to `0` or `null` but I don't see why anyone would, given that it's already disabled by default.
I guess I'm wondering what sets anonymous sessions apart that made them a big enough concern where we had a special lifespan defined for them. Are they created more often than regular sessions? Will they never be cleaned up? (Sorry for the naive questions, it's been a long time since I looked at the session mgmt code!)

I am wondering about the use case of someone who has a Kibana instance in 7.16 with anonymous access enabled, then upgrades it to 8.0 and gets annoyed at the session lifespan for their authenticated users so they disable it. Now perhaps without realizing it they have started a ticking time bomb.

If sessions are _never_ cleaned up unless they have a lifespan, then perhaps we should consider treating the anonymous provider differently in 8.0 -- e.g., leaving it with a 30 day lifespan that you have to configure independently from `xpack.security.session.lifespan: 0`. Idk if that makes sense though, just spitballing here!

Reply:
> I assume you mean 7.x?

Nope, I mean `master` and `7.x`, we're not changing this behavior in this PR.

> I guess I'm wondering what sets anonymous sessions apart that made them a big enough concern where we had a special lifespan defined for them. Are they created more often than regular sessions? Will they never be cleaned up? (Sorry for the naive questions, it's been a long time since I looked at the session mgmt code!)

The main difference is that it's super easy to create anonymous sessions without even realizing it, by _anyone_. But at the same time we don't want to hide the fact that they exist, and would like to [educate administrators about their existence](https://www.elastic.co/guide/en/kibana/7.x/kibana-authentication.html#anonymous-access-session) and implications as we're planning to rely on these for more advanced use cases in the future.

> I am wondering about the use case of someone who has a Kibana instance in 7.16 with anonymous access enabled, then upgrades it to 8.0 and gets annoyed at the session lifespan for their authenticated users so they disable it. Now perhaps without realizing it they have started a ticking time bomb.

Hmm, I don't see any issue here to be honest, that was their explicit choice, not the one we made for them by default. And we provide tools to help them if it ever becomes a problem: https://www.elastic.co/guide/en/kibana/7.x/session-management-api-invalidate.html:
```http
$ curl -X POST api/security/session/_invalidate
{
  "match" : "query",
  "query": {
    "provider" : { "type": "anonymous" }
  }
}
```

> If sessions are never cleaned up unless they have a lifespan, then perhaps we should consider treating the anonymous provider differently in 8.0 -- e.g., leaving it with a 30 day lifespan that you have to configure independently from xpack.security.session.lifespan: 0. Idk if that makes sense though, just spitballing here!

I honestly want to stop treating anonymous sessions differently unless there is a significant value in doing so (e.g. idle timeout doesn't provide much value to anonymous users, but at the same time can be very annoying.)

>  Idk if that makes sense though, just spitballing here!

Nah, let's discuss! There are as many opinions as there are people, especially when we're talking about their, people, experience :slightly_smiling_face:

Reply:
Just for the records: we discussed this with @jportner over Slack and agreed that the best option would be to improve the docs and make consequences of the disabled lifespan/idleTimeout clearer, handled here https://github.com/elastic/kibana/pull/106061/commits/7894aa51088b1068ba33293c8ef85af1a0c79f08